contracts/batcher/TwoWayBatcher.sol

```diff
@@ -0,0 +1,85 @@
+//SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.13;
+
+import "hardhat/console.sol";
+import "@equilibria/root/number/types/UFixed18.sol";
+import "@equilibria/root/token/types/Token18.sol";
+import "@equilibria/root/token/types/Token6.sol";
+import "@equilibria/root/control/unstructured/UReentrancyGuard.sol";
+import "./Batcher.sol";
+
+contract TwoWayBatcher is UReentrancyGuard, Batcher {
+    using UFixed18Lib for UFixed18;
+    using Token18Lib for Token18;
+    using Token6Lib for Token6;
+
+    event USDCLoaned(address indexed depositor, UFixed18 amount);
+    event USDCRepaid(address indexed depositor, UFixed18 amount);
+
+    error TwoWayBatcherInvalidUSDCAmount(UFixed18 amount);
+
+    UFixed18 public usdcLoansOutstanding;
+    mapping(address => UFixed18) public depositorToUSDCLoanAmount;
+
+    constructor(IEmptySetReserve reserve, Token18 dsu, Token6 usdc)
+    Batcher(reserve, dsu, usdc)
+    { }
+
+    function loanUSDC(UFixed18 amount) external nonReentrant {
```

Comment:
I think we should avoid loan / repay verbiage for a variety of reason. maybe just `deposit / withdraw` or `supply / redeem`.

Reply:
`USDC` in the naming is also a little redundant cuz there's not an alternative action with a different currency.